compiler/src/dotty/tools/dotc/Driver.scala

```diff
@@ -132,7 +135,31 @@ class Driver {
    *                    if compilation succeeded.
    */
   def process(args: Array[String], rootCtx: Context): Reporter = {
-    val (fileNames, ctx) = setup(args, rootCtx)
+    val (fileNames0, ctx0) = setup(args, rootCtx)
+    val (fileNames, ctx) =
+      if (ctx0.settings.fromTasty.value(ctx0)) {
+        // Resolve classpath and class names of tasty files
+        val (classPaths, classNames) = fileNames0.map { name =>
+          val path = Paths.get(name)
+          if (!name.endsWith(".tasty")) ("", name)
+          else if (Files.exists(path)) {
+            TastyFileUtil.getClassName(path) match {
+              case Some(res) => res
+              case _ =>
+                ctx0.error(s"Could not load classname from $name.")
+                ("", name)
+            }
+          } else {
+            ctx0.error(s"File $name does not exist.")
+            ("", name)
+          }
+        }.unzip
+        val ctx1 = ctx0.fresh
+        val classPaths1 = classPaths.distinct.filter(_ != "")
+        val fullClassPath = (ctx1.settings.classpath.value(ctx1) :: classPaths1).mkString(java.io.File.pathSeparator)
```

Comment:
I would put classPaths1 in front of the regular classpath.

Reply:
Yes, it should